Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/NodeFactory.php

```diff
@@ -175,6 +196,7 @@ public function mapNodeRowsToNodeAggregate(
                 // ... so we handle occupation exactly once ...
                 $nodesByOccupiedDimensionSpacePoints[$occupiedDimensionSpacePoint->hash] = $this->mapNodeRowToNode(
                     $nodeRow,
+                    WorkspaceName::forLive(), // todo use workspace name in content graph api
```

Comment:
`NodeAggregates` also need to operate on workspace names, but for that we would have to adjust the content graph api. That might go in a separate pr.

```
public function findNodeAggregateById(
    WorkspaceName $workspaceName,
    NodeAggregateId $nodeAggregateId
): ?NodeAggregate;
```

the cs id would be looked up internally  via the `workspaces` table.
And the cache would be similar flushed on `markStale` as the cached `subgraphs`

Reply:
The problem with this would be that the write site heavily relies on the `ContentGraph` and must work on a specific contentstream id (see `ContentStreamIdOverride`)

unless we create a custom `ContentGraph` which will use our content stream, or by actually hacky checking against the state in `ContentStreamIdOverride` we wont get around that.